packages/expo-image-picker/ios/MediaHandler.swift

```diff
@@ -23,38 +23,41 @@ internal struct MediaHandler {
 
   @available(iOS 14, *)
   internal func handleMultipleMedia(_ selection: [PHPickerResult], completion: @escaping (ImagePickerResult) -> Void) {
-    var results: [SelectedMediaInfo] = []
+    var results: [String: SelectedMediaInfo] = [:]
 
     let dispatchGroup = DispatchGroup()
     let dispatchQueue = DispatchQueue(label: "expo.imagepicker.multipleMediaHandler")
 
-    let resultHandler = { (result: SelectedMediaResult) -> Void in
+    let resultHandler = { (assetId: String?, result: SelectedMediaResult) -> Void in
       switch result {
       case .failure(let exception):
         return completion(.failure(exception))
       case .success(let mediaInfo):
         dispatchQueue.async {
-          results.append(mediaInfo)
+          results[assetId!] = mediaInfo
```

Comment:
50/50 😅 shouldn't we handle the case when the assetId is not available?

Reply:
It is always available, because we initialize picker with a photo library:
```swift
let config = PHPickerConfiguration(photoLibrary: PHPhotoLibrary.shared())
```

But I'm reworking it anyway in a subsequent PR (returning asset IDs, permission management) so there won't be that force unwrap any longer 😉